src/main/java/software/amazon/documentdb/jdbc/calcite/adapter/DocumentDbRules.java

```diff
@@ -786,9 +789,68 @@ private static Operand translateCurrentTimestamp(final Instant currentTime) {
         @SneakyThrows
         private static Operand translateDateAdd(final RexCall call, final List<Operand> strings) {
             verifySupportedDateAddType(call.getOperands().get(1));
+
+            // Is date addition between literals (including CURRENT_DATE)?
+            final boolean isLiteralCandidate = isDateLiteralCandidate(call, strings);
+            if (isLiteralCandidate) {
+                // Perform in-memory calculation before sending to server.
+                return getDateAddLiteralOperand(strings);
+            }
+            // Otherwise, perform addition on server.
             return new Operand("{ \"$add\":" + "[" + Util.commaList(strings) + "]}");
         }
 
+        private static boolean isDateLiteralCandidate(final RexCall call, final List<Operand> strings) {
+            final boolean allLiterals = call.getOperands().stream()
+                    .allMatch(op -> {
+                        final SqlKind opKind = op.getKind();
+                        final String opName = op.toString();
+                        return opKind == SqlKind.LITERAL
+                                || opName.equalsIgnoreCase(CURRENT_DATE);
+                    });
+            final boolean allHaveQueryValue = strings.stream().allMatch(op -> op.getQueryValue() != null);
+            return allLiterals && allHaveQueryValue;
+        }
+
+        private static Operand getDateAddLiteralOperand(final List<Operand> strings) {
+            final String queryValue0 = strings.get(0).getQueryValue();
+            final String queryValue1 = strings.get(1).getQueryValue();
+            final BsonDocument document0 =  BsonDocument.parse("{field: " + queryValue0 + "}");
+            final BsonDocument document1 =  BsonDocument.parse("{field: " + queryValue1 + "}");
+
+            long sum = 0L;
+            for (BsonValue v : new BsonValue[]{document0.get("field"), document1.get("field")}) {
+                switch (v.getBsonType()) {
+                    case DOUBLE:
+                        sum += v.asDouble().longValue();
+                        break;
+                    case STRING:
```

Comment:
Can these cases occur? Do we need unit tests for them? I think it caused a slightest of drops in branch coverage

Reply:
Removed data types that won't be found.